src/frontend/src/pages/TeamsPage/TeamMembersView.tsx

```diff
@@ -0,0 +1,123 @@
+  const [isEditingMembers, setIsEditingMembers] = useState(false);
+  const [members, setMembers] = useState(team.members);
+
+  const { isLoading, isError, error, data } = useAllUsers();
+  const {
+    isLoading: memberIsLoading,
+    isError: memberIsError,
+    error: memberError,
+    mutateAsync
+  } = useSetTeamMembers(team.teamId);
+  if (isError) return <ErrorPage message={error?.message} />;
+  if (memberIsError) return <ErrorPage message={memberError?.message} />;
+  if (isLoading || memberIsLoading || !data) return <LoadingIndicator />;
+
+  const handleSubmit = async () => {
+    await mutateAsync(members.map((member) => member.userId));
+    setIsEditingMembers(false);
+  };
+
+  const hasPerms = () => {
+    return !(
+      auth.user &&
+      (auth.user.role === 'ADMIN' || auth.user.role === 'APP_ADMIN' || auth.user.userId === team.leader.userId)
+    );
+  };
+  const userToAutocompleteOption = (user: User): { label: string; id: number } => {
+    return { label: `${fullNamePipe(user)}`, id: user.userId };
+  };
+
+  if (!isEditingMembers)
+    return (
+      <PageBlock
+        title={'People'}
+        headerRight={<IconButton disabled={hasPerms()} onClick={() => setIsEditingMembers(true)} children={<Edit />} />}
+      >
+        <Grid container spacing={1}>
+          <Grid item xs={12} md={12}>
+            <b style={{ display: 'inline' }}>Lead: </b>
+            <p style={{ display: 'inline' }}>{fullNamePipe(team.leader)}</p>
+          </Grid>
+          <Grid item xs={12} md={12}>
+            <b style={{ display: 'inline' }}>Members: </b>
+            <p style={{ display: 'inline' }}>
+              {team.members
+                .map((member) => {
+                  return fullNamePipe(member);
+                })
+                .join(', ')}
+            </p>
+          </Grid>
+        </Grid>
+      </PageBlock>
+    );
+  else {
+    return (
+      <PageBlock
+        title={'People'}
+        headerRight={
+          <div style={{ display: 'flex' }}>
+            <Button onClick={() => setIsEditingMembers(false)}>Cancel</Button>
+            <Button
+              sx={{
+                ml: 2,
+                backgroundColor: theme.palette.success.main,
+                color: theme.palette.success.contrastText
+              }}
+              onClick={handleSubmit}
+            >
+              Save
+            </Button>
+          </div>
+        }
+      >
+        <Grid container spacing={1}>
+          <Grid item xs={12} md={12}>
+            <b style={{ display: 'inline' }}>Lead: </b>
+            <p style={{ display: 'inline' }}>{fullNamePipe(team.leader)}</p>
+          </Grid>
+          <Grid item xs={12} md={12}>
+            <Autocomplete
+              multiple
+              id="tags-standard"
+              //first filters the options to only include users who have not been selected to be on the team, then sorts the options by alphabetical order of their first name
+              options={data
```

Comment:
autocomplete wants something in the form `{label: string; id: string/number/whatever}`. check out how i refactored the set roles autocomplete to map the users list to be that basically. then you can do the comparison thingy and i think it should get rid of console warnings

Reply:
I made that change but am still getting console warnings maybe I implemented it incorrectly.